src/main/java/org/sagebionetworks/bridge/reporter/worker/UploadsReportGenerator.java

```diff
@@ -0,0 +1,49 @@
+package org.sagebionetworks.bridge.reporter.worker;
+
+import static java.util.stream.Collectors.counting;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.joda.time.DateTime;
+import org.joda.time.LocalDate;
+
+import org.sagebionetworks.bridge.reporter.helper.BridgeHelper;
+import org.sagebionetworks.bridge.reporter.request.ReportType;
+import org.sagebionetworks.bridge.rest.model.ReportData;
+import org.sagebionetworks.bridge.rest.model.Study;
+import org.sagebionetworks.bridge.rest.model.Upload;
+
+public class UploadsReportGenerator implements ReportGenerator {
```

Comment:
I added a test of the UploadsReportGenerator separate from these processor tests. I did not move any of these tests because they are also testing the processor's iteration over studies, the saving of the reports, etc. So it seemed useful to leave these here.